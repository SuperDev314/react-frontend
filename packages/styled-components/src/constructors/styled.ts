import constructWithOptions from './constructWithOptions';
import StyledComponent from '../models/StyledComponent';
import domElements from '../utils/domElements';

import { Target } from '../types';

type StyledTemplateFunction = ReturnType<typeof constructWithOptions>;

type ElementShortcuts = {
  [key in keyof JSX.IntrinsicElements]?: StyledTemplateFunction;
};

interface Styled extends ElementShortcuts {
  (tag: Target): StyledTemplateFunction;
}

const styled: Styled = (tag: Target) => constructWithOptions(StyledComponent, tag);

// Shorthands for all valid HTML Elements
domElements.forEach(domElement => {
  styled[domElement] = styled(domElement);
});

export default styled;
