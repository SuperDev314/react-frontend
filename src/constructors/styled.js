// @flow
import constructWithOptions from './constructWithOptions'
import type { Target } from '../types'
import domElements from '../utils/domElements'

export default (styledComponent: Function) => {
  const styled = (tag: Target) => constructWithOptions(styledComponent, tag)

  // Shorthands for all valid HTML Elements
  domElements.forEach(domElement => {
    styled[domElement] = styled(domElement)
  })

  return styled
}
