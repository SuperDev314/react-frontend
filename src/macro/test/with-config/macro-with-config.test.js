import pluginTester from 'babel-plugin-tester';
import plugin from 'babel-plugin-macros';

const exampleCode = `
import styled from '../../../macro'

styled.div\`
  background: red;
\`
`;

pluginTester({
  title: 'macro with config',
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: {
    'should not add componentId with a config disabling ssr': exampleCode,
  },
});
