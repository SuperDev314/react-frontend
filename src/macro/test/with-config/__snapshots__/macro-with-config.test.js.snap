// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macro with config should not add componentId with a config disabling ssr: should not add componentId with a config disabling ssr 1`] = `
"
import styled from '../../../macro'

styled.div\`
  background: red;
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import _styled from 'styled-components';


_styled.div.withConfig({
  displayName: 'macro-with-configtest'
})(['background:red;']);
"
`;
